### ISSUE TITLE
Slerp instead of lerp

### ISSUE THREAD

**Issue by ljleb** [ljleb/prompt-fusion-extension#46]
During interpolation, the embeddings norm becomes smaller than either of the interpolated embeddings. We should instead consider rotating the features embeddings along an arc of the 2D spiral passing by the origin and any 2 embeddings to be interpolated.

To illustrate this idea, here's a 2D slice of how interpolation is done as of right now:

![image](https://user-images.githubusercontent.com/32277961/233461975-df2d4628-bf1e-4e0c-8821-e36527cb6d8f.png)

and here's a 2D slice of how interpolation should probably be done:

![image](https://user-images.githubusercontent.com/32277961/233462485-29005313-a495-4238-9609-ba883ab6fc95.png)

TL;DR: we need to pass the interpolation ratio through trigonometric functions and then normalize the length of the resulting vector.

**Comment by ljleb** [comment]
I ran some tests and curving around the origin does not yield better results. Now trying to rotate around the negative prompt.

**Comment by ljleb** [comment]
Rotating around the empty cond definitely seems to give more coherent results than rotating around 0.

**Comment by ljleb** [comment]
I just learned this concept actually has a name: [slerp](https://en.wikipedia.org/wiki/Slerp). This needs more tests to find which of the following points for the center of the circle give the best results:

- 0
- empty cond
- negative embedding (in the direction of CFG)
